app/mailers/patient_mailer.rb

```diff
@@ -154,13 +162,13 @@ def assessment_voice(patient)
                max_retries_message: I18n.t('assessments.phone.max_retries_message', locale: lang),
                thanks: I18n.t('assessments.phone.thanks', locale: lang) }
 
+    update_last_assessment_reminder_sent(patient) # Update last send attempt timestamp before Twilio call
     if TwilioSender.start_studio_flow(patient, params)
       add_success_history(patient, patient)
     else
       add_fail_history_voice(patient)
+      clear_last_assessment_reminder_sent(patient) # Reset send attempt timestamp on failure
     end
-    # Always update the last contact time so the system does not try and send emails again.
-    patient.update(last_assessment_reminder_sent: DateTime.now)
   end
 
   def assessment_email(patient)
```

Comment:
Wouldn't we want a check in `assessment_email` as well for the race condition?
```
    # Cover potential race condition where multiple messages are sent for the same monitoree.
    return unless last_assessment_reminder_sent_eligible?(patient)
```

Reply:
Yes, good catch!